### DISCUSSION TITLE
Is it possible to set the formatter to ignore some rules?

### DISCUSSION THREAD

**Discussion by IrishWhiskey** [astral-sh/ruff#11143] [General]
Hello,

I want to ignore rule `501` when formatting as I believe it sometimes leads to less readable code. I see that the linter has the `ignore` config option, but it does not seem to work for the formatter. Is there a way to make it work?

**Comment by tjkuson** [comment]
`E501` is a linting rule and does not affect the formatter per se. Both the formatter and the `E501` lint rule use [`line-length`](https://docs.astral.sh/ruff/settings/#line-length) setting. The lint rule uses the setting to raise a message if it finds a line of code exceeding the given length, and the formatter uses the setting as to format lines such that they remain within the given setting (though sometimes lines will overrun). You can't disable line-length formatting whilst using the formatter. But, you can set the `line-length` value to a larger number so that it rarely triggers.

**Reply by IrishWhiskey** [reply]
I get it, thanks
